### PR TITLE
CLI Unit tests

### PR DESCRIPTION
**Pull request by insightmind**
This PR includes the unit tests for the CLI project

### CODE REVIEW

File: application/Core/CLI/Commands/Processing/ProcessOptions.cs [hunk 1]
```diff
@@ -1,9 +1,10 @@
 ﻿using CommandLine;
+using MORR.Shared.Utility;
```
Comment (sjend) [inline application/Core/CLI/Commands/Processing/ProcessOptions.cs]: Unused and can be removed.

File: application/Core/CLI/Commands/Record/RecordCommand.cs [hunk 2]
```diff
@@ -2,50 +2,84 @@
...
 
             try
             {
-                OutputFormatter.IsVerbose = options.IsVerbose;
+                consoleFormatter.IsVerbose = options.IsVerbose;
 
                 // Load Configuration File
-                OutputFormatter.PrintDebug(loadedFileMessage);
+                consoleFormatter.PrintDebug(loadedFileMessage);
                 var configPath = new FilePath(Path.GetFullPath(options.ConfigPath));
```
Comment (sjend) [inline application/Core/CLI/Commands/Record/RecordCommand.cs]: This is now unused and could be removed.

File: application/Core/CLI/Commands/Record/RecordOptions.cs [hunk 3]
```diff
@@ -1,9 +1,10 @@
 ﻿using CommandLine;
+using MORR.Shared.Utility;
```
Comment (sjend) [inline application/Core/CLI/Commands/Record/RecordOptions.cs]: Unused and can be removed.

File: application/Core/CLI/Commands/Validate/ValidateOptions.cs [hunk 4]
```diff
@@ -1,9 +1,10 @@
 ﻿using CommandLine;
+using MORR.Shared.Utility;
```
Comment (sjend) [inline application/Core/CLI/Commands/Validate/ValidateOptions.cs]: Unused and can be removed.

File: application/Core/CLI/Program.cs [hunk 5]
```diff
@@ -1,7 +1,14 @@
-﻿using CommandLine;
+﻿using System.IO;
+using CommandLine;
 using MORR.Core.CLI.Commands.Processing;
 using MORR.Core.CLI.Commands.Record;
 using MORR.Core.CLI.Commands.Validate;
+using MORR.Core.CLI.Interactive;
+using MORR.Core.CLI.Output;
+using MORR.Core.CLI.Utility;
+using MORR.Core.Configuration;
```
Comment (sjend) [inline application/Core/CLI/Program.cs]: Unused and can be removed.

File: application/Core/CLI/Output/IConsoleFormatter.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+namespace MORR.Core.CLI.Output
+{
+    /// <summary>
+    /// The IConsoleFormatter abstracts from the default
+    /// Console output allowing to insert different behaviors on printing
+    /// and a hierarchical printing options.
+    /// </summary>
+    public interface IConsoleFormatter
```
Comment (Gordin508) [inline application/Core/CLI/Output/IConsoleFormatter.cs]: We might have found use for a similar interface for logging purposes (same set of methods, just without `Read()`), but this is out of scope here as such an Interace would have to reside outside of the CLI package.

File: application/Core/CLITest/Commands/ProcessCommandTest.cs [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+            };
+
+            /* WHEN */
+            var returnCode = command.Execute(options);
+
+            /* THEN */
+
+            // We test if the command was successful and returned code 0.
+            Assert.AreEqual(successCode, returnCode);
+            managerMock.Verify(manager => manager.Process(It.IsAny<IEnumerable<DirectoryPath>>()), Times.Exactly(1));
```
Comment (Gordin508) [inline application/Core/CLITest/Commands/ProcessCommandTest.cs]:
```suggestion
            managerMock.Verify(manager => manager.Process(It.IsAny<IEnumerable<DirectoryPath>>()), Times.Once);
```
For consistency

File: application/Core/CLITest/Commands/ValidateCommandTest.cs [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+            var command = new ValidateCommand(managerMock.Object, outputMock.Object, bootstrapperMock.Object);
+
+            /* WHEN */
+            var returnCode = command.Execute(null);
+
+            /* THEN */
+
+            // We test if the command was successful and returned code 0.
+            Assert.AreEqual(-1, returnCode);
+            managerMock.Verify(manager => manager.LoadConfiguration(It.IsAny<FilePath>()), Times.Exactly(0));
```
Comment (Gordin508) [inline application/Core/CLITest/Commands/ValidateCommandTest.cs]:
```suggestion
            managerMock.Verify(manager => manager.LoadConfiguration(It.IsAny<FilePath>()), Times.Never);
```

File: application/Core/CLI/Output/IConsoleFormatter.cs (hunk 6 above)
Comment (insightmind) [reply application/Core/CLI/Output/IConsoleFormatter.cs]: Is this fine for you if I merge it with the interface included in the CLI?

Comment (Gordin508) [reply application/Core/CLI/Output/IConsoleFormatter.cs]: Should be fine if that makes it easier, i would suggest doing that in a new PR however.
